src/components/scenes/WcConnectionsScene.tsx

```diff
@@ -63,11 +63,15 @@ export const WcConnectionsScene = (props: Props) => {
   const onScanSuccess = async (qrResult: string) => {
     setConnecting(true)
     try {
-      let proposal: Web3WalletTypes.SessionProposal | undefined = sessionProposal[qrResult]
+      let proposal = sessionProposal.get(qrResult)
 
       if (proposal == null) {
-        proposal = await walletConnect.initSession(qrResult)
-        setSessionProposal({ [qrResult]: proposal })
+        const newProposal = await walletConnect.initSession(qrResult)
+        setSessionProposal(proposals => {
+          proposals.set(qrResult, newProposal)
```

Comment:
When storing a map in state, you need to avoid modifying the original object, but rather a copy:

```ts
const out = new Map(proposals)
out.set(qrResult, newProposal)
return out
```

Otherwise rendering could get messed up, as React doesn't see that anything changed.